src/biokbase/narrative/jobs/job.py

```diff
@@ -77,65 +117,104 @@ def __init__(self, **kwargs):
                 batch_tag: app tag
                 batch_size: number of jobs being run
         params (dict): input parameters
-        owner (str): the user who started the job
+        user (str): the user who started the job
         run_id (str): unique run ID for the job
         tag (str): the application tag (dev/beta/release)
 
         """
-        if kwargs.get("job_id", None) is None:
-            raise ValueError("Cannot create a job without a job ID!")
-        else:
-            self.job_id = kwargs.get("job_id")
-
-        # parent job: set the batch_id
-        if kwargs.get("batch_job", False) is True:
-            kwargs["batch_id"] = self.job_id
-
-        for key, value in JOB_DEFAULTS.items():
-            setattr(self, key, kwargs.get(key, value))
+        # reconstruct the ee2 job state object
+        ee2_state = kwargs.get("ee2_state", {})
+        job_input = ee2_state.get("job_input", {})
+        narr_cell_info = job_input.get("narrative_cell_info", {})
 
-        ee2_state = kwargs.get("ee2_state", None)
-        if not ee2_state:
-            ee2_state = {
-                "job_id": self.job_id,
+        narr_cell_info.update(
+            {key: kwargs[key] for key in NARR_CELL_INFO_ATTRS if key in kwargs and key not in narr_cell_info}
```

Comment:
I'm not sure if I agree here. If I'm reading this right, this says that if the command is something like:
```
ee2_state = {
    "job_input": {
        "app_id": "SomeApp"
    }
}
new_job = Job.from_attributes(ee2_state=ee2_state, app_id="SomeOtherApp", params={...whatever parameters...})
```

Then this will result in `job_input` being
```
{
    "app_id": "SomeApp",  (not "SomeOtherApp")
    "params": {... from the kwarg input...}
}
```

So, I have 2 questions.
1. Are there cases where the more nested arguments that would be in job_input anyway would show up as a kwarg?
2. If a separate kwarg for, say, app_id is given, should that override what's given in job_input if they disagree?

And, I guess a third - are there cases where they both show up and disagree with each other? 

I'm honestly not sure about the second one. I think it's probably fine right now to assume that if an `ee2_state` is given with a `job_input` structure, complete or not, it should be taken as The Truth. I'm just a unsure whether this is maybe a little overengineered or if it's solving cases that I'm not expecting.

Either way, in the interest of deadlines, I think this is fine, but maybe something to revisit.

Reply:
:sweat_smile: Yes, it is overengineered and solving cases that don't appear anywhere except in generating test data. Usually the `ee2_state` isn't in the picture when initializing a `Job` from kwargs.